docs/guides/nodejs/survey-application.mdx

```diff
@@ -0,0 +1,305 @@
+})
+```
+
+## Step 3: Generate PDF Receipts
+
+Listen for submitted events and generate a formatted PDF receipt from the stored data.
+
+```ts title:src/services/pdfs.ts
+import { receipts, submissions, output } from '../resources/resources'
+import { buildReceipt } from '../form/receipt'
+
+receipts.subscribe(async (ctx) => {
+  const { id } = ctx.req.json()
+  const submission = await submissions.get(id)
+
+  if (!submission) {
+    console.error(`No submission found for ID: ${id}`)
+    return
+  }
+
+  // Build the PDF buffer from the submission data
+  const buffer = await buildReceipt(submission)
+
+  // Store the PDF file in the bucket
+  const file = output.file(`${id}.pdf`)
+  await file.write(buffer)
+
+  console.log(`Receipt stored for ${id}`)
+})
+```
+
+### Build the PDF Receipt
+
+```ts title:src/forms/form/receipt.ts
+import PDFDocument from 'pdfkit'
+import { Submission } from './schema'
+
+export const buildReceipt = async (data: Submission) => {
+  const receipt = new PDFDocument({ bufferPages: true })
+
+  const doneWriting = new Promise<Buffer>((resolve) => {
+    const buffers: Uint8Array[] = []
+
+    // Collect PDF output in memory as chunks
+    receipt.on('data', buffers.push.bind(buffers))
+    receipt.on('end', () => {
+      const pdfData = Buffer.concat(buffers)
+      resolve(pdfData)
+    })
+
+    const { name, rating, feedback } = data
+
+    // Write header
+    receipt.font('Times-Roman').fontSize(20).text('Survey - Receipt', 100, 100)
+
+    // Write section title
+    receipt
+      .font('Times-Roman')
+      .fontSize(16)
+      .text('Primary Applicant Details', 100, 150)
+
+    // Write submission details
+    receipt
+      .font('Times-Roman')
+      .fontSize(12)
+      .text(
+        `Name: ${name}
+Rating: ${rating}
+Feedback: ${feedback || 'N/A'}`,
+        100,
+        175,
+      )
+
+    // Finalize the PDF
+    receipt.end()
+  })
+
+  return await doneWriting
+}
+```
+
+This PDF output is fairly plain, you can enhance it further using layout templates or branding.
+
+## Step 4: Delivery Logic
+
+Simulate or perform delivery of the receipt (e.g. via email or other downstream systems).
+
+```ts title:src/services/deliver.ts
+import { receipts } from '../resources/resources'
+
+receipts.subscribe(async (ctx) => {
+  const { id } = ctx.req.json()
+
+  // Handle delivery or hook into a real email/SaaS integration
+  console.log(`Delivering receipt for submission: ${id}`)
+})
+```
+
+## Step 5: Retrieve Receipts
+
+Create an endpoint that returns a download URL for the generated receipt file.
+
+```ts title:src/apis/receipts.ts
+import { api } from '@nitric/sdk'
+import { output } from '../resources/resources'
+
+const receiptApi = api('receipts')
+
+receiptApi.get('/receipts/:id', async (ctx) => {
+  const id = ctx.req.params.id
+  const file = output.file(`${id}.pdf`)
+  const url = await file.getDownloadUrl()
+  ctx.res.body = url
+})
+```
+
+## Run and Test Locally
+
+```bash
+nitric start
```

Comment:
This doesn't work - the original service `services/api.ts` still exists (and runs), but all the other services are either in `src/services` or `src/apis` according to the instructions and they won't run.

Reply:
Updating the `nitric.yaml` file like this works, although it's probably better to delete the original services folder or use it and not update the nitric.yaml.

```yaml
  - basedir: ""
    match: src/services/*.ts
    runtime: node
    start: npm run dev:services $SERVICE_PATH
  - basedir: ""
    match: src/apis/*.ts
    runtime: node
    start: npm run dev:services $SERVICE_PATH
```

Reply:
Ah yup, the instructions should be updated to remove services and add the new files into the same folder.